purchase_ux/models/purchase_order.py

```diff
@@ -39,24 +38,17 @@ class PurchaseOrder(models.Model):
 
     @api.depends(
         'state', 'order_line.qty_received', 'order_line.product_qty',
-        'manually_set_received')
-    def _get_received(self):
+        'force_delivered_status')
+    def _compute_delivery_status(self):
         precision = self.env['decimal.precision'].precision_get(
             'Product Unit of Measure')
         for order in self:
-            # on v9 odoo consider done with no more to purchase, PR has been
-            # deny, if we change it here we should change odoo behaviour on
-            # purchase orders
-            # al final dejamos  nuestro criterio porque es confuso para
-            # clientes y de hecho odoo, a diferencia de lo que dice el boton
-            # si te deja crear las facturas en done
-            # if order.state != 'purchase':
             if order.state not in ('purchase', 'done'):
                 order.delivery_status = 'no'
```

Comment:
> order.delivery_status = 'no'
 
use update()

Reply:
why update? if only one value and one record is updated then seams ok for me, I'm  missing something?

Reply:
@jjscarafia because is inside a for loop and will try to write on iteration for each record. Do you agreed?